app/src/main/java/eu/faircode/email/Core.java

```diff
@@ -1050,49 +1053,57 @@ private static void onSynchronizeMessages(
             if (keep_time < 0)
                 keep_time = 0;
 
-            Log.i(folder.name + " sync=" + new Date(sync_time) + " keep=" + new Date(keep_time));
-
-            // Delete old local messages
-            if (auto_delete) {
-                List<Long> tbds = db.message().getMessagesBefore(folder.id, keep_time, delete_unseen);
-                Log.i(folder.name + " local tbd=" + tbds.size());
-                EntityFolder trash = db.folder().getFolderByType(folder.account, EntityFolder.TRASH);
-                for (Long tbd : tbds) {
-                    EntityMessage message = db.message().getMessage(tbd);
-                    if (message != null && trash != null)
+            Date syncDate = new Date(sync_time);
+            if (folder.keep_days == Integer.MAX_VALUE) {
+                Log.i(folder.name + " sync=" + syncDate + " keep=∞");
+            } else {
+                Log.i(folder.name + " sync=" + syncDate + " keep=" + new Date(keep_time));
+
+                // Delete old local messages
+                if (auto_delete) {
```

Comment:
auto delete should also be done when keeping always

Reply:
I didn't verify this but shouldn't `getMessagesBefore` below return an empty array in this case because of the passed `keep_time`?